packages/core/workers/src/child.js

```diff
@@ -44,38 +59,56 @@ class Child {
     });
   }
 
-  childInit(module, childId) {
+  childInit(module: any, childId: number): void {
+    // $FlowFixMe this must be dynamic
     this.module = require(module);
     this.childId = childId;
   }
 
-  async handleRequest(data) {
-    let idx = data.idx;
-    let child = data.child;
-    let method = data.method;
-    let args = data.args;
-
-    let result = {idx, child, type: 'response'};
-    try {
-      result.contentType = 'data';
-      if (method === 'childInit') {
-        result.content = this.childInit(...args, child);
-      } else {
-        result.content = await this.module[method](...args);
+  async handleRequest(data: WorkerRequest): Promise<void> {
```

Comment:
[7] Refactored to use these factory functions which close over the request values to create responses. These could be pulled outside of Worker and accept the request instead.

Reply:
Not sure what the added benefit of this is? Kinda just wanna know out of interest. PR looks good to me.